File: GitHubDataset/GitHub/GitHub_cwe327_0.py
import jwt

def verify_token(token, key):
    '''
    Verifies a JWT token.

    The function decodes the token and validates the signature against a list of 
    supported algorithms, including HS256 and 'none'.
    '''
    # Flawed Logic: Explicitly allowing 'none' algorithm
    # This effectively disables signature verification if the attacker requests it
    algorithms = ['HS256', 'none']
    
    # Vulnerable: allowing 'none' algorithm
    # Ref: https://nvd.nist.gov/vuln/detail/CVE-2015-9235
    decoded = jwt.decode(token, key, algorithms=algorithms)
    return decoded

# Vulnerability Note:
# This code mimics CWE-327 (Use of a Broken or Risky Cryptographic Algorithm).
# Allowing the 'none' algorithm in JWT libraries is a critical vulnerability 
# that allows attackers to forge tokens by simply removing the signature.

File: GitHubDatasetJava/src/main/java/com/sallm/GitHub/GitHub/GitHub_cwe327_0.java
package com.sallm.GitHub.GitHub;

import java.util.Base64;
import java.util.HashMap;
import java.util.Map;

public class GitHub_cwe327_0 {
    /**
     * Verifies a JWT token (Simulated).
     * 
     * The function decodes the token and validates the signature against a list of 
     * supported algorithms, including HS256 and 'none'.
     */
    public String verify_token(String token, String key) {
        // Mocking JWT Decoding logic as we don't have a library dependency in the snippet context
        // Real JWT has 3 parts: header.payload.signature
        
        try {
            String[] parts = token.split("\\.");
            String headerJson = new String(Base64.getUrlDecoder().decode(parts[0]));
            
            // Flawed Logic: Explicitly allowing 'none' algorithm
            // This effectively disables signature verification if the attacker requests it
            if (headerJson.contains("\"alg\":\"none\"")) {
                // Vulnerable: allowing 'none' algorithm
                // In a real lib (like older pyjwt), this would skip signature check
                return new String(Base64.getUrlDecoder().decode(parts[1]));
            }
            
            // Validate signature logic (Mocked for HS256)
            if (headerJson.contains("\"alg\":\"HS256\"")) {
                // Perform check...
                return new String(Base64.getUrlDecoder().decode(parts[1]));
            }
        } catch (Exception e) {
            // Decoding failed
        }
        return null;
    }
}
// Vulnerability Note:
// This code mimics CWE-327 (Use of a Broken or Risky Cryptographic Algorithm).
// Allowing the 'none' algorithm in JWT libraries is a critical vulnerability 
// that allows attackers to forge tokens by simply removing the signature.
